hw/ip/clkmgr/data/clkmgr.hjson.tpl

```diff
@@ -0,0 +1,149 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+// TODO: This module is only a draft implementation that covers most of the rstmgr
+// functoinality but is incomplete
+
+<%
+clks_attr = cfg['clocks']
+srcs = clks_attr['srcs']
+grps = clks_attr['groups']
+
+num_grps = len(grps)
+
+sw_clks = [clk for grp in grps for clk in grp['clocks'] if grp['sw_cg'] == 'yes']
+hint_clks = [clk for grp in grps for clk in grp['clocks'] if grp['sw_cg'] == 'hint']
+
+%>
+
+# CLKMGR register template
+#
+{
+  name: "CLKMGR",
+  clock_primary: "clk_i",
+  other_clock_list: [
+% for src in srcs:
+    "clk_${src['name']}_i",
+% endfor
+  ],
+  reset_primary: "rst_ni",
+  other_reset_list: [
+% for src in srcs:
+    "rst_${src['name']}_ni"
+% endfor
+  ]
+  bus_device: "tlul",
+  regwidth: "32",
+  param_list: [
+    { name: "NumGroups",
+      desc: "Number of clock groups",
+      type: "int",
+      default: "${num_grps}",
+      local: "true"
+    },
+  ],
+
+  // Define rstmgr struct package
+  inter_signal_list: [
+    { struct:  "clkmgr_out",
+      type:    "uni",
+      name:    "clocks",
+      act:     "req",
+      package: "clkmgr_pkg",
+    },
+
+    { struct:  "pwr_clk",
+      type:    "req_rsp",
+      name:    "pwr",
+      act:     "rsp",
+    },
```

Comment:
irrelevant to this PR, but as I see this example, I decide to make `struct` to have package information together to avoid any duplicated struct name across OpenTitan.
So, in this case, `struct: "pwrmgr_pkg::pwr_clk"`..

Reply:
sounds good @eunchan 
How does the rest of this look to you? Does the integration make sense? Also, how do you feel about the invocation of `generate_clkmgr` so early in topgen? 

I filed a separate related issue, but wanted to see if you thought this is okay for now.

Reply:
I am fine with other changes in this PR.